cpp-rt.hpp

```diff
@@ -0,0 +1,97 @@
+#ifndef CPPRT_HPP
+#define CPPRT_HPP
+
+#include <stddef.h>
+#include <malloc.h>
+#include <cstdint>
+#include <cassert>
+
+#define FATAL(s) { fprintf(stderr, s); exit(1); }
+
+// This is a hack for .SO to work when they are compiled against a program/library that uses libstdc++
+extern "C" {
+static inline __attribute__((noreturn)) void __phantom_handler(void) { while(1); }
+
+void __cxa_pure_virtual (void) __attribute__ ((weak, alias("__phantom_handler")));
+void __cxa_deleted_virtual (void) __attribute__ ((weak, alias("__phantom_handler")));
+}
+
+inline void* operator new(size_t, void* p) noexcept { return p; }
+
+namespace BmCpp {
+
+struct BaseAllocation {
+    inline void*    operator new(size_t len) noexcept { return malloc(len); }
```

Comment:
I personally knew about disabling exceptions on the compiler level, but I'm not too sure what `noexcept` does  here. This might require some slight documentation for the curious reader.